formbricks/formbricks
# feat: improve configurability for upload storage

## What does this PR do?

- Improved AWS environment support by making all S3 environment variables optional other than bucket.
- Added support for custom uploads directory via the UPLOADS_DIR environment variable.




Context: 
https://discord.com/channels/979077669410979880/1212076001039884288



## How should this be tested?




1. Added temporary log statements when getting the S3 client instance to print the current configuration.
2. Launched the Launched the app with multiple S3 environment variable variations to ensure that S3 client configuration is being picked up from the environment when not explicitly set.
3. Attempted to upload background images to survey steps to trigger and verify the s3 configuration logs.

- Launched the app both with and without `UPLOAD_DIRS` set in `.env` to ensure that default behavior is maintained and that the new custom path works when specified.

## Checklist



### Required


### Appreciated
